status/condition_set.go

```diff
@@ -237,3 +239,60 @@ func (c ConditionSet) findUnhealthyDependents() []Condition {
 	})
 	return conditions
 }
+
+// objectImpl is an embedded struct type that embeds the upstream type
+type objectImpl struct {
+	unstructured.Unstructured
+}
+
+func FromUnstructured(nodeClass *unstructured.Unstructured) Object {
+	return &objectImpl{Unstructured: *nodeClass}
+}
+
+func (o *objectImpl) GetConditions() []Condition {
+	var conditions []Condition
+	c, found, err := unstructured.NestedSlice(o.Object, "status", "conditions")
```

Comment:
I wonder if we should validate this as part of the constructor. That way you don't have a chance to fail silently later. You get pre-validated that the path exists and that there are conditions present